app/src/main/java/woowacourse/shopping/presentation/cart/CartViewModel.kt

```diff
@@ -0,0 +1,250 @@
+package woowacourse.shopping.presentation.cart
+
+import android.os.Handler
+import android.os.Looper
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.map
+import com.example.domain.datasource.DataResponse
+import com.example.domain.datasource.map
+import com.example.domain.datasource.onFailure
+import com.example.domain.datasource.onSuccess
+import com.example.domain.model.Quantity
+import com.example.domain.repository.CartRepository
+import com.example.domain.repository.OrderRepository
+import com.example.domain.repository.RecentProductRepository
+import woowacourse.shopping.common.Event
+import woowacourse.shopping.common.emit
+import woowacourse.shopping.presentation.products.ProductCountActionHandler
+import woowacourse.shopping.presentation.products.uimodel.ProductUiModel
+import kotlin.concurrent.thread
+
+class CartViewModel(
+    private val recommendRepository: RecentProductRepository,
+    private val cartRepository: CartRepository,
+    private val orderRepository: OrderRepository,
+) : ViewModel(), CartActionHandler, ProductCountActionHandler {
+    private val _cartUiState = MutableLiveData<CartUiState>()
+    val cartUiState: LiveData<CartUiState> get() = _cartUiState
+
+    private val _changedCartEvent = MutableLiveData<Event<Unit>>()
+    val changedCartEvent: LiveData<Event<Unit>> get() = _changedCartEvent
+
+    val cartItemSelectedCount: LiveData<Int>
+        get() = cartUiState.map { it.cartUiModels.count { cartUiModel -> cartUiModel.isSelected } }
+
+    val cartItemAllSelected: LiveData<Boolean>
+        get() =
+            cartUiState.map { it.cartUiModels.all { cartUiModel -> cartUiModel.isSelected } }
+
+    val totalPrice: LiveData<Int>
+        get() =
+            cartUiState.map { cartUiState ->
+                cartUiState.cartUiModels.filter {
+                    it.isSelected
+                }.sumOf { it.product.price * it.quantity.count }
```

Comment:
uiModel 내부에서 스스로 계산하도록 바꿔봐도 좋겠네요 ㅎㅎ
```suggestion
                }.sumOf { uiModel -> uiModel.getTotalPrice() }
```